apps/api/e2e/utils/test-helpers.ts

```diff
@@ -37,29 +39,43 @@ export function createTestClient(baseUrl?: string): ApiClient {
  * Register a new team and return a client configured with its API credentials
  */
 export async function registerTeamAndGetClient(
-  adminClient: ApiClient,
+  adminApiKey: string,
   teamName?: string,
   email?: string,
   contactPerson?: string,
+  walletAddress?: string,
 ) {
   // Ensure database is initialized
   await ensureDatabaseInitialized();
 
+  const sdk = getApiSdk(adminApiKey);
   // Register a new team
-  const result = await adminClient.registerTeam(
-    teamName || `Team ${generateRandomString(8)}`,
-    email || `team-${generateRandomString(8)}@test.com`,
-    contactPerson || `Contact ${generateRandomString(8)}`,
-  );
-
-  if (!result.success || !result.team) {
+  const result = await sdk.admin.postApiAdminTeamsRegister({
+    teamName: teamName || `Team ${generateRandomString(8)}`,
+    email: email || `team-${generateRandomString(8)}@test.com`,
+    contactPerson: contactPerson || `Contact ${generateRandomString(8)}`,
+    walletAddress: walletAddress || generateRandomEthAddress(),
+    metadata: {},
+  });
+
+  if (!result.success || !result.team || typeof result.team.id !== "string") {
     throw new Error("Failed to register team");
   }
 
   // Create a client with the team's API key
   const client = new ApiClient(result.team.apiKey);
-
-  return { client, team: result.team, apiKey: result.team.apiKey };
+  // TODO: work on replacing `ApiClient` instances with SDK instances everywhere
+  // TODO: the return for this function can be cleaned up when we use the sdk everywhere
```

Comment:
I wanted to use the sdk in this helper, but to keep this PR from getting even bigger I kept the return type the same.